src/nlp.jl

```diff
@@ -71,6 +71,20 @@ mutable struct NLPData
     evaluator
 end
 
+"""
+    nlp_objective_function(model::Model)
+
+Returns the nonlinear objective function or `nothing` if no nonlinear objective
+function is set.
+"""
+function nlp_objective_function(model::Model)
+    if model.nlp_data isa Nothing
```

Comment:
Not sure if we have a style point on `x isa Nothing` or `x === nothing`?

Reply:
I have always wondered what was the recommended way. Is there one way that is better recognized by the compiler ?

Reply:
I've always used `x === nothing`. It seems simpler and I know that it's fast. Not sure about `x isa Nothing`.

Reply:
Julia 1.1 introduces `isnothing`: https://github.com/JuliaLang/julia/pull/29679